### ISSUE TITLE
[Bug] TorchMaster Feral Flare light sources breaking ability to copy

### ISSUE THREAD

**Issue by Fladorius** [Direwolf20-MC/BuildingGadgets#150]
Issue: When trying to copy an area that has Feral Flare light sources, the copy and paste tool will not select any area that includes a light source. If I select the floor of the structure, then try and extend it upward using the shift+right click menu, it will extend the box, but doesn't say area copied. When putting the tool into the manager after this, the image only shows the initial copy.

[Example Structure](https://i.imgur.com/NU6qxqL.png) I'm trying to copy
[Result](https://i.imgur.com/MQAogL2.png) after copying the floor and extending it up using shift+right click to the top level just before the actual feral flare

Versions:
Forge: 14.23.5.2796
Building Gadgets: 2.4.6
Torch Master: 1.6.4.62

**Comment by MajorTuvok** [comment]
Are they TileEntities? (Though there should have been an error message if they were...?)

**Comment by MichaelHillcox** [comment]
I think I may have also had this issue although not known it was related to the lanterns

**Comment by Fladorius** [comment]
I'm not 100% sure if I'm reading TorchMaster code correctly, but the [Feral Flare Lantern](https://github.com/Xalcon/TorchMaster/blob/ded382169e0f39b0091a25322e2c0abf09d324ae/src/main/java/net/xalcon/torchmaster/common/tiles/TileEntityFeralFlareLantern.java#L115) itself is a tile entity, and it spawns [InvisibleLightSources](https://github.com/Xalcon/TorchMaster/blob/master/src/main/java/net/xalcon/torchmaster/common/blocks/BlockInvisibleLight.java), which are just blocks. I'm not sure if I can test in game if it is a tile entity or not as there isn't hit box(? sorry not sure the terminology) that shows up when looking at it like there would be when there looking at another block, or a flower, so I can't select it with the Building or Exchanging gadgets.

There isn't any error message, it just refuses to select the area, or it won't copy the area using the shift+right click menu to update the selected area. Normally there aren't any error messages when trying to copy an area with tile entities. Trying to copy this [structure](https://i.imgur.com/StzRlxx.png) with a furnace in the center will copy everything successfully except the furnace as I would expect.

**Comment by Fladorius** [comment]
I investigated the logs and found a related exception log. Diving in to the code, it looks like Direwolf already fixed the issue in commit [82bfa99](https://github.com/Direwolf20-MC/BuildingGadgets/commit/82bfa99e7c225281f3ec9baab719f4df0e67aa17#diff-e9ab5a316a948bcd314511458fff484dR141) in BlockMapIntState.java a few days ago. I built the source to test and it looks like it works now with the invisible blocks. Going to close this since it's already been fixed.
